index.js

```diff
@@ -15,46 +39,135 @@ module.exports = function(parent, theirs, mine) {
     return [];
   }
 
-  return recurse(parent, theirs, mine);
+  return recurse(parent, theirs, mine, [], options);
 };
 
-function recurse(parent, theirs, mine, path) {
+/**
+ * Recursively traverse the objects to handle 3-way-diff
+ *
+ * @param {object} parent - Object with parent values
+ * @param {object} theirs - Object with their changes
+ * @param {object} mine - Object with my changes
+ * @param {array} path - Path to value in parent/theirs/mine
+ * @param {object} options - Key specific options for diff
+ * @returns {Array} - Array of objects specifying the differences
+ */
+function recurse(parent, theirs, mine, path, options) {
   var path = path || [];
   var results = [];
 
   // Handle all the keys that exist in mine
   _.forOwn(mine, function(value, key) {
-    if(!_.isArray(value) && !_.isObject(value)) {
+    results = results.concat(processKeyValuePair(key, value, parent, theirs, mine, path, options));
+  });
+
+  // Handle all the keys that exist in parent but not mine
+  _.forOwn(parent, function(value, key) {
+    // Skip all keys that exist in mine
+    if (_.isUndefined(mine[key])) {
+      results = results.concat(processKeyValuePair(key, value, parent, theirs, mine, path, options));
+    }
+  });
+
+  return results;
+}
+
+/**
+ * Determine process in which key/value should be compared
+ *
+ * @param key - The current object key to dif
+ * @param value - The current object value to diff
+ * @param {object} parent - Object with parent values
+ * @param {object} theirs - Object with their changes
+ * @param {object} mine - Object with my changes
+ * @param {array} path - Path to value in parent/theirs/mine
+ * @param {object} options - Key specific options for diff
+ * @returns {Array} - Array of objects specifying the differences
+ */
+function processKeyValuePair(key, value, parent, theirs, mine, path, options) {
+  var results = [];
+
+  // Only process keys that have no options, or have not been flagged as ignored
+  if (_.isUndefined(options[key]) || (!_.isUndefined(options[key]) && !options[key]['ignoreKey'])) {
+    if (!_.isArray(value) && !_.isObject(value)) {
       path.push(key);
-      var differences = compareValues(parent[key], theirs[key], mine[key], path);
-      if(differences) {
+      var differences = compareValues(parent[key], theirs[key], mine[key], path, options[key] || {});
+      if (differences) {
         results = results.concat(differences);
       }
       path.pop();
     }
-    else if(_.isArray(value)) {
-      // TODO
+    else if (_.isArray(value)) {
+      var array = value;
+      // If single dimension array run the comparison
+      if (value.filter(function (i) {
+          return !_.isArray(i) && !_.isObject(i);
+        }).length == value.length) {
```

Comment:
This is a pretty complex `if` statement. Should probably pull this out to a variable